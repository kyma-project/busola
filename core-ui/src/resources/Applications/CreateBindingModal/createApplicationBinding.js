export function createApplicationBinding(application, namespace, services) {
  return {
    apiVersion: 'applicationconnector.kyma-project.io/v1alpha1',
    kind: 'ApplicationMapping',
    metadata: {
      name: application.metadata.name,
      namespace,
    },
    spec: { services },
  };
}
