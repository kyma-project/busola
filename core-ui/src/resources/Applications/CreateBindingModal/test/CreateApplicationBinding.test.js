import { createApplicationBinding } from '../createApplicationBinding';

describe('createApplicationBinding', () => {
  it('forms valid binding', () => {
    const application = { metadata: { name: 'test-application-name' } };
    const namespace = 'test-namespace';
    const services = ['service1', 'service2'];
    expect(
      createApplicationBinding(application, namespace, services),
    ).toMatchSnapshot();
  });
});
