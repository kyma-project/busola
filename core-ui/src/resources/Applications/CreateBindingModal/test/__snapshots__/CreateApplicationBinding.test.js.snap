// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createApplicationBinding forms valid binding 1`] = `
Object {
  "apiVersion": "applicationconnector.kyma-project.io/v1alpha1",
  "kind": "ApplicationMapping",
  "metadata": Object {
    "name": "test-application-name",
    "namespace": "test-namespace",
  },
  "spec": Object {
    "services": Array [
      "service1",
      "service2",
    ],
  },
}
`;
